aquadoggo/src/replication/service.rs

```diff
@@ -264,17 +273,18 @@ impl ConnectionManager {
         self.send_service_message(ServiceMessage::ReplicationFailed(peer));
     }
 
+    /// Generates our new announcement state we can then propagate to all known and future peers.
+    async fn update_announcement(&mut self) {
+        let target_set = self.target_set().await;
+        self.announcement = Some(Announcement::new(target_set));
+    }
+
     /// Determine if we can attempt new replication sessions with the peers we currently know
     /// about.
     async fn update_sessions(&mut self) {
         // Determine the target set our node is interested in
         let target_set = self.target_set().await;
 
-        if let Err(err) = target_set.validate() {
-            warn!("Not initiating replication: {err}");
-            return;
-        }
```

Comment:
I would assume that our own target set is always valid (especially when it was created through the `new` constructor) or are there any cases I overlooked?